### ISSUE TITLE
Pendantic mode?

### ISSUE THREAD

**Issue by mikepurvis** [ros-infrastructure/rosinstall_generator#31]
Right now, if you use `--deps`, and rosinstall_generator can't find a dependency, it silently drops it from the produced workspace. It would be great to have a flag which makes it fail with an angry message and error code when this occurs.

**Comment by wjwwood** [comment]
You can try to make a pull request for it, but my initial impression is that we have to skip unknown packages because otherwise you'd never be able to deal with dependencies which are not ROS packages.

I don't think we'll have anytime to spend on this soon, so I'll put it in the `untargeted` milestone.

**Comment by mikepurvis** [comment]
It would be great to tie in rosdep in order to resolve those, but I can understand wanting to avoid an unnecessary dependency.

Where would I start poking at this? The code is a bit hard to follow; I'm not seeing an explicit point where it drops unknown packages.

**Comment by wjwwood** [comment]
I'm not familiar with it either. @dirk-thomas might be able to point you in the right direction since he originally wrote this.

**Comment by dirk-thomas** [comment]
Sorry for being late to this ticket.

The stack to consider only released ROS packages looks like this:
* https://github.com/ros-infrastructure/rosinstall_generator/blob/8dce72299628b479891c6bfba6a016aeef899a98/src/rosinstall_generator/generator.py#L345
* https://github.com/ros-infrastructure/rosinstall_generator/blob/8dce72299628b479891c6bfba6a016aeef899a98/src/rosinstall_generator/distro.py#L83 (notice the argument `ros_packages_only=True`)
* https://github.com/ros-infrastructure/rosdistro/blob/8b13881c13bfa48255554f260ae5dbe87541bc14/src/rosdistro/dependency_walker.py#L59-L60

**Comment by dirk-thomas** [comment]
I will go ahead and close this ticket since there is no plan to work on this (hence the milestone untargeted in the past). Any contribution to support this use case is welcome though.
